### PR TITLE
Add section on dropping DelayedInit

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

File: docs/docs/reference/dropped/delayedInit.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    object HelloWorld extends App {
+      println("Hello, world!")
+    }
+
+However, the code is now run in the initializer of the object, which on
+some JVM's means that it will only be interpreted. So, better not use it
+for benchmarking! Also, if you want to access the command line arguments,
+you need to use an explicit `main` method for that.
+
+    object Hello extends App {
```
Comment (smarter) [inline docs/docs/reference/dropped/delayedInit.md]: In this example, is there any point in doing `extends App`?

**Comment by smarter** [comment]
See also https://github.com/lampepfl/dotty/issues/559 which lead to the creation `dotty.runtime.LegacyApp`.

File: docs/docs/reference/dropped/delayedInit.md (hunk 1 above)
Comment (odersky) [inline docs/docs/reference/dropped/delayedInit.md]: No, good catch.

**Comment by odersky** [comment]
#559 does not seem to be an issue anymore, at least not for simple cases where we do not access `args`. I added a test.

**Comment by olafurpg** [comment]
Thanks @smarter I opened https://github.com/scalacenter/scalafix/issues/174 to handle the cases when `args` is accessed. Currently, that case compiles but `args == null` at runtime. What do you think about removing `args` to get a compile-time error instead?

https://github.com/lampepfl/scala/blob/7ecfce1fb8d39275f082aaa3ad4dc0eee197391c/src/library/scala/App.scala#L46

**Comment by smarter** [comment]
We can't really remove it because it comes from scala-library, and we don't recompile scala-library currently in dotty, we use the one from 2.11.11 as-is.

**Comment by olafurpg** [comment]
I see, that makes sense @smarter

File: docs/docs/reference/dropped/type-projection.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+title: Dropped: General Type Projection
+---
+
+Scala so far allowed general type projection `T#A` where `T` is an arbitrary type
+and `A` names a type member of `T`.
+
+Dotty allows this only if `T` is a class type. The change was made because
+unrestricted type projection is [unsound](https://github.com/lampepfl/dotty/issues/1050).
+
+The restriction rule out the [type-level encoding of compbinator
```
Comment (ShaneDelmore) [inline docs/docs/reference/dropped/type-projection.md]: /s/compbinator/combinator

**Comment by olafurpg** [comment]
We just added the Scalafix `NoExtendsApp` rewrite that replaces `extends App` with an explicit `main` function, regardless if `args` is used or not https://scalacenter.github.io/scalafix/#NoExtendsApp

**Pull request merged**